### ISSUE TITLE
SOTA - Librispeech s2s training - Learning rate not increasing

### ISSUE THREAD

**Issue by omprakashsonie** [facebookresearch/wav2letter#669]
Hi,
The learning rate didn't warmup to 0.03 after 64,000 updates.

Thank you for your help in advance.

<img width="822" alt="image" src="https://user-images.githubusercontent.com/8159628/83343628-8c819080-a31a-11ea-8256-aac9c062fc26.png">

Here is the configuration file:
--runname=am_transformer_seq2seq_librispeech-29may9am
--rundir=/home/omprakash.s/sota
--archdir=/root/wav2letter/recipes/models/sota/2019
--arch=am_arch/am_transformer_s2s.arch
--tokensdir=/home/omprakash.s/sota/am
--tokens=librispeech-train-all-unigram-10000.tokens
--lexicon=/home/omprakash.s/sota/am/librispeech-train+dev-unigram-10000-nbest10.lexicon
--train=/home/omprakash.s/sota/lists/train-clean-100.lst,/home/omprakash.s/sota/lists/train-clean-360.lst,/home/omprakash.s/sota/lists/train-other-500.lst
--valid=dev-clean:/home/omprakash.s/sota/lists/dev-clean.lst,dev-other:/home/omprakash.s/sota/lists/dev-other.lst
--criterion=transformer
--mfsc
--usewordpiece=true
--wordseparator=_
--am_decoder_tr_dropout=0.1
--am_decoder_tr_layerdrop=0.1
--am_decoder_tr_layers=6
--maxdecoderoutputlen=120
--labelsmooth=0.05
--dataorder=output_spiral
--inputbinsize=25
--attnWindow=softPretrain
--softwstd=4
--trainWithWindow=true
--pretrainWindow=3
--attention=keyvalue
--encoderdim=256
--memstepsize=5000000
--eostoken=true
--pcttraineval=1
--pctteacherforcing=99
--sampletarget=0.01
--netoptim=adagrad
--critoptim=adagrad
--lr=0.03
--lrcrit=0.03
--linseg=0
--momentum=0.0
--maxgradnorm=0.1
--onorm=target
--sqnorm
--nthread=6
--batchsize=8
--filterbanks=80
--minloglevel=0
--logtostderr
--enable_distributed

**Comment by omprakashsonie** [comment]
Can anyone pls help?

**Comment by tlikhomanenko** [comment]
@omprakashsonie 

Can you point to the commit of w2l you are using? We had some recent fixes, so could you confirm that you have this error with the latest master?

**Comment by samin9796** [comment]
@tlikhomanenko 
I am also facing the same issue. I set warmup to 8000 but even after 100000 updates, lr is not increasing. I am using the docker image created 4 weeks ago.

**Comment by tlikhomanenko** [comment]
@samin9796 

could you add your config file?

**Comment by omprakashsonie** [comment]
Thanks Tatiana for your input.
I think I had taken w2l code on 23-May.

**Comment by tlikhomanenko** [comment]
The fix is on the way, thanks for flagging!

**Comment by samin9796** [comment]
@tlikhomanenko 
Here is the config file: 
[001_config.txt](https://github.com/facebookresearch/wav2letter/files/4719596/001_config.txt)

**Comment by syhw** [comment]
This is my fault, I sent the fix 2 days after I introduced the bug but then I forgot about it and didn't merge it into master. It's merged now: https://github.com/facebookresearch/wav2letter/commit/e7d9e89760affd4d3292d6010447a96b54fdab88. I am sorry about that, thanks @tlikhomanenko for bringing this issue to my attention and pushing for me to close it.
